pkg/wal/processor/translator/wal_translator.go

```diff
@@ -231,10 +234,18 @@ func (t *Translator) fillEventMetadata(event *wal.Data, log *schemalog.LogEntry,
 			continue
 		}
 
-		if t.versionFinder != nil && t.versionFinder(col, tbl) && !foundVersion {
-			foundVersion = true
-			event.Metadata.InternalColVersion = col.PgstreamID
-			continue
+		if t.versionFinder != nil && !foundVersion {
```

Comment:
Instead of checking in a few places if `versionFinder` is nil, can we not return early at the beginning if it is nil and remove the checks?

Reply:
Ah, nevermind, I see you still need to check by id too.

Reply:
Yeah, the version finder is optional, but we reuse the same loop for finding the identity as well, which is required, to avoid having two loops over the table columns. 
The version finder is a hook to introduce custom functionality when identifying the version column. In the case of xata, we have our own logic to identify the version, and this update helps the transition towards using the LSN instead, with feature flag granularity.

Reply:
We also don't want to default to using the LSN if a version finder is provided and no version is found, since that would mean events would be replicated using the LSN, whereas it should be treated as an invalid event from the user point of view. This is why we only check if the version is found when the version finder is provided (hence the second check on the versionFinder you mentioned).